### PR TITLE
Add support for SSL policy

### PR DESCRIPTION
**Pull request by jlubawy**
Depends on https://github.com/gruntwork-io/terraform-google-load-balancer/pull/15

### CODE REVIEW

**Comment by romankor** [comment]
@jlubawy any chance you guys merge any of the PR's ?!

**Comment by jlubawy** [comment]
> @jlubawy any chance you guys merge any of the PR's ?!

I'm not a member of the gruntwork team, but would also like to see these related PRs merged, they seem simple enough.

File: modules/http-load-balancer-website/variables.tf [hunk 1]
```diff
@@ -53,6 +53,12 @@ variable "ssl_certificate" {
   default     = ""
 }
 
+variable "ssl_policy" {
+  description = "Link to ssl policy. Omitted if not provided"
+  type        = string
+  default     = ""
```
Comment (robmorgan) [inline modules/http-load-balancer-website/variables.tf:59]:
```suggestion
  default     = null
```
Let's set the default to `null` for better consistency with our other modules.

File: modules/http-load-balancer-website/variables.tf [hunk 2]
```diff
@@ -53,6 +53,12 @@ variable "ssl_certificate" {
   default     = ""
 }
 
+variable "ssl_policy" {
+  description = "Link to ssl policy. Omitted if not provided"
```
Comment (robmorgan) [inline modules/http-load-balancer-website/variables.tf:57]:
```suggestion
  description = "A reference to the SslPolicy resource that will be associated with the TargetHttpsProxy resource. If not set, the TargetHttpsProxy resource will not have any SSL policy configured."
```
We'll adopt the TF provider docs as its clearer to the users.

**Review by robmorgan — CHANGES_REQUESTED** [review]
This PR should also have an explicit test case for the `ssl_policy` functionality.

**Comment by jlubawy** [comment]
> This PR should also have an explicit test case for the `ssl_policy` functionality.

@robmorgan what would the test case cover in your mind? I'm not sure I have the bandwidth to add a test case at the moment (I don't have much experience here), but I'd like to help get this PR merged.

**Comment by robmorgan** [comment]
Hi @jlubawy, I'm also short on bandwidth, but am happy to review any contributions. We need a new test case to explicitly pass in an `ssl_policy` parameter so we can properly test this feature. Our tests are written using [Terratest](https://terratest.gruntwork.io). I imagine the best place to start would likely be upstream in the [terraform-google-load-balancer](https://github.com/gruntwork-io/terraform-google-load-balancer/) repo. I'd start by maybe adding a `ssl_test.go` file then copying the code from another test case. This new test case should ensure the `ssl_policy` has a value set when deploying the example.

**Pull request closed without merging**